huh_scriven_velocity.py

```
@@ -31,48 +33,86 @@ def Ux(Theta, Phi):
+    Returns:
+        tuple: (Theta_grid, Phi_grid, Ux_rel_grid, Uy_rel_grid)
+    """
+    # Set matplotlib backend based on GUI parameter
+    if not GUI:
+        import matplotlib
+        matplotlib.use('Agg')  # Use non-interactive backend
+        # Create output directory if it doesn't exist
+        os.makedirs(output_dir, exist_ok=True)
+    
+    # Create a meshgrid for Theta and Phi
+    Theta_grid = []
```

Pylint (3.3.7)

[warning] Redefining name 'Theta_grid' from outer scope (line 113)

(W0621)